src/workerd/api/queue.c++

```diff
@@ -87,8 +169,16 @@ kj::Promise<void> WorkerQueue::sendBatch(
     // TODO(perf): We should be able to encode the data directly into bodyBuilder's buffer to
     // eliminate a lot of data copying (whereas now encodeBase64 allocates a new buffer of its own
     // to hold its result, which we then have to copy into bodyBuilder).
-    bodyBuilder.addAll(kj::encodeBase64(serializedBodies[i]));
-    bodyBuilder.addAll("\"}"_kj);
+    bodyBuilder.addAll(kj::encodeBase64(serializedBodies[i].body));
```

Comment:
Yeah, this is a bit of a bummer

Reply:
Improving this whole mess feels like another `TODO(perf)` to throw on the pile for the future (where "the pile" is largely the internal ticket titled `Reduce number of times we copy messages in sendBatch in the C++ binding`).

Maybe this means that Kenton has successfully brought me to the dark side over the years, but this does all feel like it would be both cleaner and much more efficient if used capnproto to deliver these batches :)